src/topology/order/lattice.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2021 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import topology.basic
+import topology.constructions
+import topology.algebra.ordered.basic
+
+/-!
+# Topological lattices
+
+In this file we define mixin classes `has_continuous_inf` and `has_continuous_sup`. We define the
+class `topological_lattice` as a topological space and lattice `L` extending `has_continuous_inf`
+and `has_continuous_sup`.
+
+## References
+
+* [Gierz et al, A Compendium of Continuous Lattices][GierzEtAl1980]
+
+## Tags
+
+topological, lattice
+-/
+
+/--
+Let `L` be a topological space and let `L×L` be equipped with the product topology and let
+`⊓:L×L → L` be an infimum. Then `L` is said to have *(jointly) continuous infimum* if the map
+`⊓:L×L → L` is continuous.
+-/
+class has_continuous_inf (L : Type*) [topological_space L] [has_inf L] : Prop :=
+(continuous_inf : continuous (λ p : L × L, p.1 ⊓ p.2))
+
```

Comment:
You may want to add a lemma `continuous.inf`, by analogy with `continuous.mul`.

Reply:
Yes, I'm working towards showing that the bounded continuous maps from a topological space to a normed lattice ordered group form a normed lattice ordered group (subject to a technical condition) so I would need this in the next PR. I've added it to this PR.